packages/platform-browser/src/dom/shared_styles_host.ts

```diff
@@ -36,18 +36,54 @@ export class SharedStylesHost {
 export class DomSharedStylesHost extends SharedStylesHost implements OnDestroy {
   // Maps all registered host nodes to a list of style nodes that have been added to the host node.
   private _hostNodes = new Map<Node, Node[]>();
+  private _styleNodesInDOM: Map<string, HTMLStyleElement>|undefined;
 
-  constructor(@Inject(DOCUMENT) private _doc: any) {
+  constructor(@Inject(DOCUMENT) private doc: Document, @Inject(APP_ID) appId: string) {
     super();
-    this._hostNodes.set(_doc.head, []);
+
+    const styles: NodeListOf<HTMLStyleElement>|undefined =
+        this.doc.head?.querySelectorAll(`style[ng-transition="${appId}"]`);
+
+    if (styles?.length) {
+      const styleMap = new Map<string, HTMLStyleElement>();
+
+      styles.forEach(style => {
+        if (style.textContent != null) {
+          styleMap.set(style.textContent, style);
```

Comment:
Should we trim the content here?
```suggestion
          styleMap.set(style.textContent.trim(), style);
```

Reply:
I don't think so, as otherwise we'd also need to `trim` when we `get` and `delete`.